Blocks/SS/signature.h

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ The MIT License (MIT)
+
+ Copyright (c) 2016 Grigory Nikolaenko <nikolaenko.grigory@gmail.com>
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+ *******************************************************************************/
+
+#ifndef BLOCKS_SS_SIGNATURE_H
+#define BLOCKS_SS_SIGNATURE_H
+
+#include "../../TypeSystem/struct.h"
+#include "../../TypeSystem/Schema/schema.h"
+
+namespace current {
+namespace ss {
+
+CURRENT_STRUCT(StreamNamespaceName) {
+  CURRENT_FIELD(exposed_namespace, std::string);
```

Comment:
Now when I think of it, these two names -- `exposed_namespace` and `top_level_name` -- are a bit too long. @mzhurovich, do you have the same concern? In this case, any suggestions?

I'd go as far as saying something as simple as `"namespace_name"` and `"{struct|type}_name"` would already be better. Obviously, these two are not the suggestions per se -- just a mere observation some rethought and simplification could help us here.